documentation/modules/con-securing-kafka-authentication.adoc

```diff
@@ -0,0 +1,115 @@
+  tls:
+    authentication:
+      type: tls
+  external:
+    type: loadbalancer
+    tls: true
+    authentication:
+      type: tls
+# ...
+----
+
+[id='con-mutual-tls-authentication-{context}']
+== Mutual TLS authentication
+
+Mutual TLS authentication is always used for the communication between Kafka brokers and ZooKeeper pods.
+
+For mutual authentication or two-way authentication, both the server and the client present certificates.
+Strimzi can configure Kafka to use TLS (Transport Layer Security) to provide encrypted communication between Kafka brokers and clients either with or without mutual authentication. When you configure mutual authentication, the broker authenticates the client and the client authenticates the broker.
+
+NOTE: TLS authentication is more commonly one-way, with one party authenticating the identity of another. For example, when HTTPS is used between a web browser and a web server, the server obtains proof of the identity of the browser.
+
+.When to use mutual TLS authentication for clients
+
+Mutual TLS authentication is recommended for authenticating Kafka clients when:
+
+* The client supports authentication using mutual TLS authentication
+* It is necessary to use the TLS certificates rather than passwords
+* You can reconfigure and restart client applications periodically so that they do not use expired certificates.
+
+[id='con-scram-sha-authentication-{context}']
+== SCRAM-SHA authentication
+
+SCRAM (Salted Challenge Response Authentication Mechanism) is an authentication protocol that can establish mutual authentication using passwords. Strimzi can configure Kafka to use SASL (Simple Authentication and Security Layer) SCRAM-SHA-512 to provide authentication on both unencrypted and TLS-encrypted client connections. TLS authentication is always used internally between Kafka brokers and ZooKeeper nodes. When used with a TLS client connection, the TLS protocol provides encryption, but is not used for authentication.
+
+The following properties of SCRAM make it safe to use SCRAM-SHA even on unencrypted connections:
+
+* The passwords are not sent in the clear over the communication channel.
+Instead the client and the server are each challenged by the other to offer proof that they know the password of the authenticating user.
+
+* The server and client each generate a new challenge for each authentication exchange.
+This means that the exchange is resilient against replay attacks.
+
+.Supported SCRAM credentials
+
+Strimzi supports SCRAM-SHA-512 only.
+When a `KafkaUser.spec.authentication.type` is configured with `scram-sha-512` the User Operator will generate a random 12 character password consisting of upper and lowercase ASCII letters and numbers.
+
+.When to use SCRAM-SHA authentication for clients
+
+SCRAM-SHA is recommended for authenticating Kafka clients when:
+
+* The client supports authentication using SCRAM-SHA-512
+* It is necessary to use passwords rather than the TLS certificates
```

Comment:
Again, I think these two are bad arguments. The first makes it sound that whenever your clients supports SCRAM-SHA-512 we recommend you to use it. The second does not offer any actual reasons. It basically says _If you know that you need to use passwords and not TLS certificates, we recommend you to use passwords_ which does nto make much sense.

Reply:
removed